internal/util/get/local.go

```diff
@@ -0,0 +1,118 @@
+package get
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt/internal/errors"
+	"github.com/GoogleContainerTools/kpt/internal/util/pkgutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func AddGcloudConfigMap(dir, pkgName string) error {
+	const op errors.Op = "kptfileutil.PullLocalGcloudConfig"
+	data, err := pullLocalGcloudConfig(pkgName)
+	if err != nil {
+		return err
+	}
+	cm := yaml.MustParse(`
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: 
+data: {}
+`)
+	cm.SetName(pkgName)
+	cm.SetDataMap(data)
+	return writeFile(filepath.Join(dir, pkgutil.LocalGloudConfigFileName(pkgName)), cm)
+}
+
+func pullLocalGcloudConfig(pkgName string) (map[string]string, error) {
+	projectID := getGcloudConfig("project")
+	if projectID == "" {
+		return nil, fmt.Errorf("`project` has not been set in `gcloud`")
+	}
+	account := getGcloudConfig("core/account")
+	if account == "" {
+		fmt.Println("`core/account` has not been set in `gcloud`")
+	}
+	zone := getGcloudConfig("compute/zone")
+	if zone == "" {
+		fmt.Println("`compute/zone` has not been set in `gcloud`")
+	}
+	region := getGcloudConfig("compute/region")
+	if region == "" {
+		fmt.Println(fmt.Errorf("`compute/region` has not been set in `gcloud`"))
+	}
+	orgID, err := getGcloudOrgID(projectID)
+	if err != nil {
+		return nil, err
+	}
+	if orgID == "" {
+		fmt.Println("`Organization` or `Folder` not found")
+	}
+
+	return map[string]string{
+		"name":      pkgName,
+		"namespace": projectID,
+		"projectID": projectID,
+		"zone":      zone,
+		"region":    region,
+		"domain":    account,
```

Comment:
This is the whole account, right, not just the domain? We need to split out the domain. Better here than in every function.

Reply:
I'm actually not sure. Domain isn't used frequently while some functions will need access to the user.

Reply:
I wouldn't expect the user's account to be used to provision resources in production.

I have a specific purpose in mind for domain, which is to use it for org-wide groups like gke-security-groups@
https://github.com/GoogleCloudPlatform/blueprints/blob/main/catalog/gke/cluster/cluster.yaml#L39
and the foundational groups in https://github.com/GoogleCloudPlatform/blueprints/tree/main/catalog/iam-foundation

We can add additional fields to the context in the future as needed.